### PR TITLE
Fix compiler warning

### PR DESCRIPTION
**Pull request by mlangc**
Fixes #1002444

### CODE REVIEW

File: org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/move/MoveClass.scala [hunk 1]
```diff
@@ -131,6 +131,7 @@ class MoveClass extends RefactoringExecutorWithWizard {
             (change :: textChanges, newFiles)
           case (change: NewFileChange, (textChanges, newFilesChanges)) =>
             (textChanges, change :: newFilesChanges)
+          case (other, _) => throw new AssertionError(s"Unexpected change $other")
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/move/MoveClass.scala]: Please improve the message in a way that it explains that only `TextChange` and `NewFileChange` make sense here.

Comment (dragos) [inline org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/move/MoveClass.scala]: I don't see any difference between throwing `MatchError` and `AssertionError`. Actually, as @sschaef noticed, `MatchError` gives you more information. I understand the warning is gone, but then let's report the error in a better way. (or use `@unchecked` to silence the warning).

**Comment by mlangc** [comment]
Hmm, please tell me if you like the new message better; if not I'd go for the `@unchecked` option.

**Comment by sschaef** [comment]
LGTM

**Pull request merged**